backend/src/apiserver/template/v2_template.go

```diff
@@ -340,37 +343,51 @@ func (t *V2Spec) validatePipelineJobInputs(job *pipelinespec.PipelineJob) error
 		if input, ok := runtimeConfig.GetParameterValues()[name]; !ok {
 			// If the parameter is optional, or there is a default value, it's ok to not have a user input
 			if !param.GetIsOptional() && param.GetDefaultValue() == nil {
-				return fmt.Errorf("parameter %s is not optional, yet has neither default value nor user provided value", name)
+				return util.NewInvalidInputError(
+					fmt.Sprintf("parameter %s is not optional, yet has neither default value nor user provided value", name),
+				)
 			}
 		} else {
 			// Verify the parameter type is correct
 			switch param.GetParameterType() {
 			case pipelinespec.ParameterType_PARAMETER_TYPE_ENUM_UNSPECIFIED:
-				return fmt.Errorf("input parameter %s has unspecified type", name)
+				return util.NewInvalidInputError(fmt.Sprintf("input parameter %s has unspecified type", name))
```

Comment:
I believe `util.NewInvalidInputError()` takes string format and arguments. So you can call `util.NewInvalidInputError("input parameter %s has unspecified type", name)` instead.

Reply:
I think you are right. I will change accordingly.